enums.rb

```diff
@@ -0,0 +1,238 @@
+        my_each do |element|
+          response = true if element.is_a?(arg)
+        end
+      elsif arg.is_a? Regexp
+        my_each do |element|
+          response = true if element.to_s.match(arg)
+        end
+
+      else
+        my_each do |element|
+          response = true if element.eql?(arg)
+        end
+      end
+
+    else
+      response = false
+      my_each do |element|
+        response = true if element
+      end
+    end
+    response
+  end
+
+  def my_none?(arg = nil)
+    response = true
+    if block_given?
+      response = true
+      my_each do |element|
+        response = false if yield(element)
+      end
+
+    elsif !arg.nil?
+
+      if arg.is_a? Class
+        my_each do |element|
+          response = false if element.is_a?(arg)
+        end
+      elsif arg.is_a? Regexp
+        my_each do |element|
+          response = false if element.to_s.match(arg)
+        end
+
+      else
+
+        my_each do |element|
+          response = false if element.eql?(arg)
+        end
+      end
+    else
+      response = true
+      my_each do |element|
+        response = false if element
+      end
+    end
+    response
+  end
+
+  def my_count(*arg)
+    count = 0
+    if block_given?
+      my_each do |element|
+        count += 1 if yield(element)
+      end
+
+    elsif !arg[0].nil?
+      my_each do |element|
+        count += 1 if element.eql?(arg[0])
+      end
+      return count
+
+    else
+
+      count = size
+    end
+    count
+  end
+
+  def my_map(proc = nil)
+    arr = []
+    if proc.equal?(nil) == false
+      my_each do |element|
+        arr << proc.call(element)
+      end
+    elsif block_given?
+      my_each do |element|
+        arr << yield(element)
+      end
+    else
+      return to_enum
+    end
+    arr
+  end
+
+  def my_inject(* args)
+    to_a if is_a? Range
+    if args[0].equal?(:+) and args[1].nil?
+      acum = 0
+      my_each do |element|
+        acum += element
+      end
+      acum
+    elsif args[0].equal?(:*) and args[1].nil?
+      acum = 1
+      my_each do |element|
+        acum *= element
+      end
+      acum
+    elsif args[0].is_a? Numeric and args[1].equal?(:+)
+      acum = args[0]
+      my_each do |element|
+        acum += element
+      end
+      acum
+
+    elsif args[0].is_a? Numeric and args[1].equal?(:*)
+      acum = args[0]
+      my_each do |element|
+        acum *= element
+      end
+      acum
+
+    elsif block_given? and !args[0].nil?
+      acum = args[0]
+      my_each do |element|
+        acum = yield(acum, element)
+      end
+      acum
+
+    elsif block_given? and args.size.eql?(0)
+      if is_a? Range
+        arr = to_a
+        acum = arr[0]
+      else
+        acum = self[0]
+      end
+      acum = 0 unless acum.is_a? String
+      my_each do |element|
+        acum = yield(acum, element)
+      end
+      acum
+    else
+      to_enum
+    end
+  end
+
+  def my_multiply()
+    my_inject(:*)
+  end
```

Comment:
[OPTIONAL]

- [ ] should have working `multiply_els`  that uses `my_inject` as described on Odin ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/multiply_els.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- This is a hard requirement so I'm very sorry this wasn't pointed out by the previous reviewers, this method should be called mutiply_els and it should take an enumerable as a parameter, also this is not an Enumerable method so it should be outside the Enumerable module or in a different module